src/app/validation/lib/architecture/statistic.ex

```diff
@@ -68,15 +87,25 @@ defmodule Architecture.Statistic do
     @impl true
     def init(params) do
       Logger.metadata(context: __MODULE__)
-      Logger.info("subscribing to log providers", process_module: __MODULE__)
-      Enum.each(params.mod.log_providers(), &LogProvider.Junction.subscribe(&1, params.resource))
+      Logger.info("subscribing to providers", process_module: __MODULE__)
+      Enum.each(params.mod.providers(),
+	fn provider ->
+	  cond do
+	    Util.has_behaviour?(provider,Architecture.Log_provider) ->
+	      &LogProvider.Junction.subscribe(&1, params.resource)
```

Comment:
This is returning the function as a value, not calling it. Replace this with `LogProvider.Junction.subscribe(provider, params.resource)` to actually call the function.